src/Network/Response.php

```diff
@@ -677,10 +739,50 @@ public function location($url = null)
     public function body($content = null)
     {
         if ($content === null) {
-            return $this->_body;
+            $this->stream->rewind();
+            $result = $this->stream->getContents();
+            if (empty($result)) {
+                return null;
+            }
+
+            return $result;
         }
 
-        return $this->_body = $content;
+        // BC compatibility
+        if (is_callable($content)) {
+            $content = $this->_handleCallableBody($content);
```

Comment:
Callbackstream - which one? Cake or Zend? Both use a different interface than the current Stream object plus you would create a new instance of the stream object here which would break the immutability of the original response?